plugin.video.ahltv/resources/lib/ahl_tv.py

```diff
@@ -0,0 +1,253 @@
+
+def color_string(string, color):
+    return '[COLOR=' + color + ']' + string + '[/COLOR]'
+
+def add_game_listitem(name, title, game_id, icon=None, fanart=None, info=None, video_info=None, audio_info=None):
+    ok = True
+
+    u = sys.argv[0] + "?url=&mode=" + str(104) + "&name=" + quote(name) + "&game_id=" + quote(str(game_id))
+
+    liz = xbmcgui.ListItem(name)
+
+    if icon is not None:
+        liz.setArt({'icon': icon, 'thumb': icon, 'fanart': fanart, 'clearlogo': CLEARLOGO})
+    else:
+        liz.setArt({'icon': ICON, 'thumb': ICON, 'fanart': fanart, 'clearlogo': CLEARLOGO})
+
+    if fanart is not None:
+        liz.setProperty('fanart_image', fanart)
+    else:
+        liz.setProperty('fanart_image', FANART)
+
+    liz.setProperty("IsPlayable", "true")
+
+    liz.setInfo(type="Video", infoLabels={"Title": title})
+    if info is not None:
+        liz.setInfo(type="Video", infoLabels=info)
+    if video_info is not None:
+        liz.addStreamInfo('video', video_info)
+    if audio_info is not None:
+        liz.addStreamInfo('audio', audio_info)
+
+    liz.setProperty('dbtype', 'video')
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=False)
+    xbmcplugin.setContent(addon_handle, 'videos')
+
+    return ok
+
+def select_game(game_id):
+    game_json = get_game_info(game_id)
+    played_id = None
+
+    if game_json['blivegameflag'] == 2 and not game_json['currentlyStreaming']: # Game is not streaming at the moment, but should be live
+        dialog = xbmcgui.Dialog()
+        dialog.ok(LOCAL_STRING(32011), LOCAL_STRING(32012))
+        sys.exit()
+    elif (game_json['blivegameflag'] <= -1 and game_json['blivegameflag'] >= -5) or (game_json['blivegameflag'] == 2 and game_json['currentlyStreaming']): # VOD game or Live game
+        dialog = xbmcgui.Dialog()
+
+        multiple_audio = game_json['multipleAudioSetting'] == 1
+        stream_info = get_game_streams(game_id, multiple_audio)
+
+        x = dialog.select(LOCAL_STRING(32013), stream_info['list_items'])
+        log('user selected stream: '+ str(x))
+
+        if x >= 0:
+            seek_secs = 0
+            selected_audio = stream_info['streams'][x]['audio_id']
+            if stream_info['streams'][x]['startSeconds'] is not None:
+                seek_secs = stream_info['streams'][x]['startSeconds']
+
+            xbmcplugin.setResolvedUrl(addon_handle, True, stream_info['list_items'][x])
+
+            while not xbmc.Player().isPlayingVideo():
+                xbmc.Monitor().waitForAbort(0.5)
+
+            # Seek to the start time of the video if it is set
+            if xbmc.Player().isPlayingVideo():
+                xbmc.sleep(250)
+                xbmc.executebuiltin('Seek(' + str(seek_secs) + ')')
+                played_id = log_played(game_id, selected_audio)
+
+            track_minute = gmtime().tm_min
+
+            while xbmc.Player().isPlayingVideo():
+                xbmc.sleep(10000)
```

Comment:
No go. You can't use xbmc.sleep with values higher than a few seconds. Please use waitforabort and add `not xbmc.Monitor().abort requested()` to the while condition

Reply:
This has been updated with the recommended changes